pkg/lease/lease.go

```diff
@@ -0,0 +1,73 @@
+package lease
+
+import (
+	"os"
+	"time"
+
+	"github.com/golang/glog"
+	coordinationv1 "k8s.io/api/coordination/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// LeaseReconciler reconciles a Secret object
+type LeaseReconciler struct {
+	KubeClient           kubernetes.Interface
+	LeaseName            string
+	LeaseDurationSeconds int32
+	componentNamespace   string
+}
+
+func (r *LeaseReconciler) Reconcile() {
+	if len(r.componentNamespace) == 0 {
+		r.componentNamespace = getPodNamespace()
+	}
+	lease, err := r.KubeClient.CoordinationV1().Leases(r.componentNamespace).Get(r.LeaseName, metav1.GetOptions{})
+
+	switch {
+	case errors.IsNotFound(err):
+		// create lease
+		lease := &coordinationv1.Lease{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      r.LeaseName,
+				Namespace: r.componentNamespace,
+			},
+			Spec: coordinationv1.LeaseSpec{
+				LeaseDurationSeconds: &r.LeaseDurationSeconds,
+				RenewTime: &metav1.MicroTime{
+					Time: time.Now(),
+				},
+			},
+		}
+		if _, err := r.KubeClient.CoordinationV1().Leases(r.componentNamespace).Create(lease); err != nil {
```

Comment:
Most likely we'll need to update the pod permissions to do this create action.

Reply:
And the update action below.

Reply:
We updated the pod permissions for managed cluster collector pods in https://github.com/open-cluster-management/klusterlet-addon-operator/pull/21/files. Do we need to update the search-chart too? @jlpadilla

Reply:
@SherinV that's all we needed. Thanks for confirming! 👍

Reply:
Could you explain why we needed to update the pod permissions for this call?

Reply:
So that the pod has create and update permissions on lease objects. Otherwise the create will fail with `pod doesn't have permission to create/update lease in this namespace`